packages/amplify-provider-awscloudformation/src/root-stack-builder/types.ts

```diff
@@ -0,0 +1,31 @@
+import * as cdk from '@aws-cdk/core';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as iam from '@aws-cdk/aws-iam';
+
+export interface AmplifyRootStackTemplate {
+  deploymentBucket?: s3.CfnBucket;
+  authRole?: iam.CfnRole;
+  unauthRole?: iam.CfnRole;
+
+  addCfnParameter(props: cdk.CfnParameterProps, logicalId: string): void;
+  addCfnOutput(props: cdk.CfnOutputProps, logicalId: string): void;
+  addCfnMapping(props: cdk.CfnMappingProps, logicalId: string): void;
+  addCfnCondition(props: cdk.CfnConditionProps, logicalId: string): void;
+  addCfnResource(props: cdk.CfnResourceProps, logicalId: string): void;
+}
+
+export interface Template {
```

Comment:
don't we have the same type defined in core? can we re-use instead of re-defining here again?